tracer/src/Datadog.Trace/ClrProfiler/NativeCallTargetDefinition.cs

```diff
@@ -72,6 +77,21 @@ internal struct NativeCallTargetDefinition
             }
 
             TargetSignatureTypesLength = (ushort)(targetSignatureTypes?.Length ?? 0);
+
+            TargetMethodArgumentsToLoad = IntPtr.Zero;
+            if (targetMethodArgumentsToLoad?.Length > 0)
+            {
+                // The Marshal operations only operate on short not ushort (CLS-compliance)
+                TargetMethodArgumentsToLoad = Marshal.AllocHGlobal(targetMethodArgumentsToLoad.Length * Marshal.SizeOf(typeof(short)));
+                var ptr = TargetMethodArgumentsToLoad;
+                for (var i = 0; i < targetMethodArgumentsToLoad.Length; i++)
+                {
+                    Marshal.WriteInt16(ptr, (short)targetMethodArgumentsToLoad[i]);
+                    ptr += Marshal.SizeOf(typeof(short));
+                }
+            }
```

Comment:
I think you should also do the disposal of the array.

Reply:
Done in 0992cfb1ea3f764bb703d8915a78be2a53bca1a3 . Does that look fine to you?